files-csharp/FastestWayToReadATextFileInCsharp/FastestWayToReadATextFileInCsharp/WaysToReadATextFileInCsharp.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Text;
+
+namespace FastestWayToReadATextFileInCsharp
+{
+    public class WaysToReadATextFileInCsharp
+    {
+        private readonly string _sampleFilePath;
+
+        public WaysToReadATextFileInCsharp(string sampleFilePath)
+        {
+            _sampleFilePath = sampleFilePath;
+        }
+
+        public string UseFileReadAllLines()
+        {
+            var stringArray = File.ReadAllLines(_sampleFilePath);
+
+            return string.Join(Environment.NewLine, stringArray);
+        }
+
+        public string UseFileReadAllText() => File.ReadAllText(_sampleFilePath);
+        
+        public string UseFileReadLines()
+        {
+            var stringArray = File.ReadLines(_sampleFilePath);
+
+            return string.Join(Environment.NewLine, stringArray);
+        }
+
+        public string UseStreamReaderReadLine()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+            var stringBuilder = new StringBuilder();
+            string fileLine;
+
+            while ((fileLine = streamReader.ReadLine()) != null)
+            {
+                stringBuilder.AppendLine(fileLine);
+            }
+
+            return stringBuilder.ToString().TrimEnd();
+        }
+
+        public string UseStreamReaderReadToEnd()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+
+            return streamReader.ReadToEnd();
+        }
+
+        public string UseStreamReaderReadBlock()
+        {
+            using var streamReader = new StreamReader(_sampleFilePath);
+            var buffer = new char[1024];
+            int numberRead;
+            var stringBuilder = new StringBuilder();
+
+            while ((numberRead = streamReader.ReadBlock(buffer, 0, buffer.Length)) > 0)
+            {
+                stringBuilder.Append(buffer, 0, numberRead);
+            }
+
+            return stringBuilder.ToString();
+        }
+
+        public string UseBufferedStreamObject()
+        {
+            var stringBuilder = new StringBuilder();
+
+            using (var fileStream = new FileStream(_sampleFilePath, FileMode.Open))
+            using (var bufferedStream = new BufferedStream(fileStream))
```

Comment:
Since we are looking at using the BufferedStream here, we should experiment with calling the FileStream constructor that disables buffering. By default FileStream will use an internal buffer of 4096 bytes. So this code ends up "double buffering".

Reply:
@cjaynjoku same here, we can add another method that doesn't do double buffering by disabled FileStream buffering. and we need to explain it in the article as well.